pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -68,10 +73,27 @@ func GetBaseDomain() (string, error) {
 		}
 		return nil
 	}); err != nil {
-		return "", errors.Wrap(err, "failed UserInput for base domain")
+		return "", "", errors.Wrap(err, "failed UserInput for base domain")
 	}
 
-	return domain, nil
+	var zoneID string
+	if zoneList, ok := publicZoneMap[domain]; ok {
+		zoneListLen := len(zoneList)
+		if zoneListLen == 1 {
+			zoneID = zoneList[0]
+		} else if zoneListLen > 1 {
```

Comment:
No need for the `else` here.  We don't build empty-slice values into `publicZoneMap`, so I'm ok with either an explicit guard:

```go
if zoneListLen == 0 {
  return "", "", errors.New("those installer people don't know what they're doing.  File a bug at...")
}
if zoneListLen == 1 {
  return domain, zoneList[0], nil
}
err = survey.AskOne(...
```

Or with the panic-if-we-break-it:

```go
if zoneListLen <= 1 {
  return domain, zoneList[0], nil // panic if we accidentally provide a way to get empty zoneList slices
}
err = survey.AskOne(...
```

Reply:
I think this is a style issue. However, I will comply ;-) Is this the normal way to handle if/else in go?